### PR TITLE
added cursor

### PR DESCRIPTION
**Pull request by vadim-frolov**

### CODE REVIEW

File: substreams-sink/src/lib.rs [hunk 1]
```diff
@@ -4,6 +4,47 @@ pub mod pb {
 use pb::{stream_client::StreamClient, Request, Response};
 use tonic::{codegen::*, Status};
 
+#[derive(Debug)]
+pub struct BlockRef {
+    pub id: String,
+    pub num: u64,
```
Comment (benjaminbollen) [inline substreams-sink/src/lib.rs:10]: should we not carry forward timestamp, and call this "clock" instead?

File: substreams-sink/src/lib.rs [hunk 2]
```diff
@@ -4,6 +4,47 @@ pub mod pb {
 use pb::{stream_client::StreamClient, Request, Response};
 use tonic::{codegen::*, Status};
 
+#[derive(Debug)]
+pub struct BlockRef {
+    pub id: String,
```
Comment (jorgeantonio21) [inline substreams-sink/src/lib.rs:9]: Can we add docs to explain what is the role of each of these fields ? Also, isn't `BlockRef` an already existing `Rust` type in another repo ? (maybe derived from some proto file ?)

File: substreams-sink/src/lib.rs [hunk 3]
```diff
@@ -4,6 +4,47 @@ pub mod pb {
...
+    pub num: u64,
+}
+
+impl BlockRef {
+    pub fn new(id: String, num: u64) -> Self {
+        Self { id, num }
+    }
+}
+#[derive(Debug)]
+pub struct Cursor {
```
Comment (jorgeantonio21) [inline substreams-sink/src/lib.rs:19]: Add some brief doc to explain the role of the `Cursor` type.

File: substreams-sink/src/lib.rs (hunk 1 above)
Comment (benjaminbollen) [reply substreams-sink/src/lib.rs:10]: resolved, let's not expand the scope of cursor at this point

**Review by benjaminbollen — APPROVED** [review]
LGTM

**Comment by jorgeantonio21** [comment]
Let's merge it for now, we can add docs later when we address #49.

Resolves #35. 
Resolves #44.

**Pull request merged**